# Instrucciones

1. Determine el modelo de regresión lineal que le permita entender la relación entre la variable Gastos y las variables regresoras Años_capacitación, Años_Cargo, Años_Compañía	y Edad.
2. ¿Qué puede concluir respecto de los estadísticos globales del modelo?
3. ¿Qué puede concluir respecto de los estadísticos individuales del modelo?

In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as ss
import statsmodels.formula.api as smf # Nos permite ajustar el modelo de regresión

In [ ]:
# Importamos la base de datos y la denominamos df por "data frame"
df = pd.read_csv('solemne_forma_3.csv', sep=';', decimal=",")

In [ ]:
df

Gastos  Años_capacitación  Años_Cargo  Años_Compañía       Edad
0     847112.0                 21           1              2  34.746119
1    1067445.0                 16           1              2  56.215982
2    2186990.0                 10           1              3  46.838813
3    3651382.0                  0           1              3  57.868950
4    1205573.0                 14           1              3  33.210502
..         ...                ...         ...            ...        ...
980  2970844.0                  8           1              3  46.329452
981  1789115.0                 11           1              2  58.399087
982  2154232.0                 13           1              3  32.516438
983   846427.0                  0           1              3  33.802055
984  2465356.0                  0           2              3  33.802055

[985 rows x 5 columns]

In [ ]:
df.describe()

Gastos  Años_capacitación  Años_Cargo  Años_Compañía        Edad
count  9.850000e+02         985.000000  985.000000     985.000000  985.000000
mean   1.523686e+06          12.395939    1.228426       2.647716   41.453666
std    1.003847e+06           8.483460    0.643608       0.735813   12.353589
min    1.000000e-29           0.000000    1.000000       1.000000   20.315753
25%    7.531140e+05           0.000000    1.000000       2.000000   31.510959
50%    1.253110e+06          14.000000    1.000000       3.000000   38.124429
75%    2.019736e+06          21.000000    1.000000       3.000000   51.602511
max    5.633724e+06          24.000000    9.000000       4.000000   74.832192

# Solución

1.

In [ ]:
# Usamos el método de los mínimos cuadrados para ajustar la regresión lineal multiple
Modelo = smf.ols('Gastos ~ Años_capacitación+Años_Cargo+Años_Compañía+Edad', df).fit() #Ajustamos el modelo
# Usamos summary para pedir el resultado del modelo
Modelo.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 Gastos   R-squared:                       0.457
Model:                            OLS   Adj. R-squared:                  0.455
Method:                 Least Squares   F-statistic:                     206.3
Date:                Thu, 18 Mar 2021   Prob (F-statistic):          2.29e-128
Time:                        02:47:10   Log-Likelihood:                -14708.
No. Observations:                 985   AIC:                         2.943e+04
Df Residuals:                     980   BIC:                         2.945e+04
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
Intercept          1.631e+06   1.74e+05      9.353      0.000    1.29e+06    1.97e+06
Años_capacitación -6.807e+04   3372.577    -20.183      0.000   -7.47e+04   -6.15e+04
Años_Cargo        -1.694e+05   3.83e+04     -4.424      0.000   -2.45e+05   -9.43e+04
Años_Compañía      1.067e+05   3.56e+04      2.994      0.003    3.68e+04    1.77e+05
Edad               1.598e+04   2033.483      7.861      0.000     1.2e+04       2e+04
==============================================================================
Omnibus:                       58.869   Durbin-Watson:                   2.075
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              138.785
Skew:                           0.335   Prob(JB):                     7.30e-31
Kurtosis:                       4.712   Cond. No.                         336.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

1. El modelo obtenido es $Gastos=1.631exp^{6}-6.807exp^{4}* Años\_capacitación-1.694exp^{5}*Años\_Cargo+1.067exp^{5}*Años\_Compañía+1.598exp^{4}*Edad$

2. Análisis estadísticos globales. A partir de los valores de F, R2 y R2 Adj podemos concluir que el modelo es capaz de explicar cerca del 45% de la varianza. F es significativo ($F_{est} < pvalor$). Luego, se rechaza la hipótesis nula de que los coeficientes de regresiones son iguales a cero, por lo que al menos un coeficiente debe ser distinto de cero.

3. Los coeficientes asociados a cada variable mantienen un test-t estadísticamente significativo ($ t_{est} < pvalor$), por lo que se rechaza la hipótesis nula para estas variables. Es decir, su coeficiente de regresión es distinto a cero.
